src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -75,12 +84,38 @@ public long amount() {
         throw new UnsupportedOperationException("amount() not yet implemented");
     }
 
-    // @todo #15:30min Implement prefix() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
-    public String prefix() {
-        throw new UnsupportedOperationException("prefix() not yet implemented");
+    public String prefix() throws Exception {
+        try {
+            final String prefix = new ItemAt<>(
+                new SplitText(
+                    new TextOf(
+                        this.transaction
+                    ),
+                    new TextOf(";")
+                ).iterator(),
+                //@checkstyle MagicNumberCheck (1 line)
+                3
```

Comment:
@paulodamaso let's explicitly check bounds of the result instead of relying on catching `Exception`. See my other comment for details

Reply:
@carlosmiranda `ItemAt` already checks the bounds of the list generated by the parsed string; there is no need to check it again.